### PR TITLE
Hugging Face Integration (Front End)

### PR DESCRIPTION
**Pull request by tarrekshaban**
This PR adds the following:
- Adds new dependencies to `package.json` for Luxon and PrettyByes for use in data presentation
- Updates API Model Files / Types based upon most recent Backend version
- Add a new HuggingFace oriented create flow
- Creates a new ButtonInput component for selecting from an enumeration of options
- Creates a new InteractiveTable component for rendering data and enabling selection from that data
- Adds Endorsed Models to the data folder
- Updates our BaseService to v1 Service and adds new endpoints for the HF flow

### CODE REVIEW

**Comment by tarrekshaban** [comment]
image

V1 of Interactive table component!

File: modelserver/tasks.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import logging
 import os
-import pdb
```
Comment (a10y) [inline modelserver/tasks.py:3]: this change is already on main, you should be able to just rebase and make this go away

File: frontend/src/pages/NewModel.tsx [hunk 2]
```diff
@@ -1,96 +1,235 @@
...
+                                title: "Import from HuggingFace",
+                                description: "Select a model from HuggingFace to deploy to your Web Server.",
+                                value: "hugging-face"
+                            },
+                        ]}
+                        onSelect={setSelection}
+                    />
+                </div>
 
+                {selection == "none" && (<React.Fragment />)}
```
Comment (a10y) [inline frontend/src/pages/NewModel.tsx:229]: nit: should use `===` when comparing strings

File: frontend/src/pages/NewModel.tsx [hunk 3]
```diff
@@ -1,96 +1,235 @@
...
+                        </div>)
+                    }
+                </>
+            )}
+        </form>
+    )
+}
+
+export default function NewModel() {
+    const [selection, setSelection] = useState("none");
```
Comment (a10y) [inline frontend/src/pages/NewModel.tsx:196]: Can we make this consistent with the other things, i.e.

```
const [selection, setSelection] = useState<string | undefined>();
```

File: modelserver/tasks.py (hunk 1 above)
Comment (tarrekshaban) [reply modelserver/tasks.py:3]: Ack, will address

File: frontend/src/pages/NewModel.tsx (hunk 3 above)
Comment (tarrekshaban) [reply frontend/src/pages/NewModel.tsx:196]: Haha see my response to your prior comment, aligned!

File: frontend/src/pages/NewModel.tsx (hunk 2 above)
Comment (a10y) [reply frontend/src/pages/NewModel.tsx:229]: for the other string comparisons below we should technically be using === as well but that sounds good to me!

**Review by a10y — APPROVED** [review]
Looks great!!

**Comment by a10y** [comment]
One other thing that might be fun: for the curated models view can we show below the description `Last updated 3 days ago` e.g. by checking the most recent commit on each of those repos.

Not for this but for a FLUP

File: modelserver/tasks.py (hunk 1 above)
Comment (tarrekshaban) [reply modelserver/tasks.py]: Done

File: frontend/src/pages/NewModel.tsx (hunk 2 above)
Comment (tarrekshaban) [reply frontend/src/pages/NewModel.tsx:229]: Addressed

File: frontend/src/pages/NewModel.tsx (hunk 3 above)
Comment (tarrekshaban) [reply frontend/src/pages/NewModel.tsx:196]: Addressed!

**Pull request merged**